plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -328,18 +328,24 @@ func (w *windowsEventLog) getRecord(evtHandle EvtHandle) (*windowsEventLogRecord
 		return nil, fmt.Errorf("utf16ToUTF8Bytes() err %v", err)
 	}
 
+	// The insertion strings could be in either EventData or UserData
```

Comment:
Is this true for all supported versions of Windows or is this a divergence with just Win11?

Reply:
From my testings, even some win11 does NOT have this issue.
(I downloaded a win11 iso from Microsoft and also got a win11 AMI from customer. With the same CWA binary, the former format message without problem but the later returns with placeholders). I feel like this could be some type of setting to the OS or windows event log, but this also means that users can change the setting and further makes CWA pushing the unformatted message (before this change)

Reply:
Based on this lengthy read I think it is valid to check `EventData` first, then check `UserData` for all of our _supported_ versions of Windows.

https://github.com/libyal/libevtx/blob/main/documentation/Windows%20XML%20Event%20Log%20(EVTX).asciidoc

>As previously mentioned the event strings (and binary data) are stored as "Data" elements in the "EventData" element within the Event XML. Another way to store the event data is in a "UserData" element. The information in this section is partially deduced on the behavior of the "General", "Details Friendly View" and "Details XML View" of Event Viewer.

>In some events the data is not stored in a "EventData" tag within the Event XML but in a "UserData" tag, e.g.